app/controllers/support/courses_controller.rb

```diff
@@ -9,11 +9,19 @@ def index
     end
 
     def edit
-      course
+      @edit_course_form = Support::EditCourseForm.new(course)
     end
 
     def update
-      if course.update(update_course_params)
+      @edit_course_form = Support::EditCourseForm.new(course)
+
+      @edit_course_form.assign_attributes(update_course_params)
+      @edit_course_form.save
+
+      @edit_course_form.valid?
+      @edit_course_form.course_valid?
+
+      if @edit_course_form.errors.none?
```

Comment:
This reads really messy

The mechanism of saving should of cause validation to occur which in turns leads to a boolean denoting has it been successful or a failure with errors

Reply:
yup would be good if this is tucked away into the form so the controller just calls `save` and handles the outcome (redirect or re-render)